From 00b2273857fdda66032f6996859f658377444e99 Mon Sep 17 00:00:00 2001
From: Hyun Kwon <hyun.kwon@xilinx.com>
Date: Tue, 13 Feb 2018 09:33:39 -0800
Subject: [PATCH 0268/1852] drm: xlnx: zynqmp: Use xlnx pipeline calls

commit 25f6bad14313e616777f4ec8df805b2ebd9e3a01 from
https://github.com/Xilinx/linux-xlnx.git

Since xlnx_drm is transformed into sort of library, new xlnx pipeline
functions should be called explicitly. Accommodate that change.

Signed-off-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/zynqmp_dpsub.c | 12 ++++++++++++
 drivers/gpu/drm/xlnx/zynqmp_dpsub.h |  1 +
 2 files changed, 13 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/zynqmp_dpsub.c b/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
index 6be1556c8603..c1a84827a781 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
+++ b/drivers/gpu/drm/xlnx/zynqmp_dpsub.c
@@ -22,6 +22,8 @@
 #include <linux/platform_device.h>
 #include <linux/pm_runtime.h>
 
+#include "xlnx_drv.h"
+
 #include "zynqmp_disp.h"
 #include "zynqmp_dp.h"
 #include "zynqmp_dpsub.h"
@@ -91,10 +93,18 @@ static int zynqmp_dpsub_probe(struct platform_device *pdev)
 		goto err_component;
 	}
 
+	dpsub->master = xlnx_drm_pipeline_init(pdev);
+	if (IS_ERR(dpsub->master)) {
+		dev_err(&pdev->dev, "failed to initialize the drm pipeline\n");
+		goto err_populate;
+	}
+
 	dev_info(&pdev->dev, "ZynqMP DisplayPort Subsystem driver probed");
 
 	return 0;
 
+err_populate:
+	of_platform_depopulate(&pdev->dev);
 err_component:
 	component_del(&pdev->dev, &zynqmp_dpsub_component_ops);
 err_disp:
@@ -108,8 +118,10 @@ static int zynqmp_dpsub_probe(struct platform_device *pdev)
 
 static int zynqmp_dpsub_remove(struct platform_device *pdev)
 {
+	struct zynqmp_dpsub *dpsub = platform_get_drvdata(pdev);
 	int err, ret = 0;
 
+	xlnx_drm_pipeline_exit(dpsub->master);
 	of_platform_depopulate(&pdev->dev);
 	component_del(&pdev->dev, &zynqmp_dpsub_component_ops);
 
diff --git a/drivers/gpu/drm/xlnx/zynqmp_dpsub.h b/drivers/gpu/drm/xlnx/zynqmp_dpsub.h
index 2c1455f288dc..6606beffee15 100644
--- a/drivers/gpu/drm/xlnx/zynqmp_dpsub.h
+++ b/drivers/gpu/drm/xlnx/zynqmp_dpsub.h
@@ -22,6 +22,7 @@
 struct zynqmp_dpsub {
 	struct zynqmp_dp *dp;
 	struct zynqmp_disp *disp;
+	struct platform_device *master;
 };
 
 #endif /* _ZYNQMP_DPSUB_H_ */
-- 
2.31.1

